Data race occurs in advanced indexing test

`TestArrayAdvancedIndexingVectorValue` does not pass locally.
The parameter `{'shape': (2, 3, 4), 'indexes': (slice(None), [[0, 1], [2, 0]]),
      'value': numpy.arange(2 * 2 * 2 * 4).reshape(2, 2, 2, 4)}` seems to cause data race.
```py
___________________________________________________________ TestArrayAdvancedIndexingVectorValue_param_3.test_adv_setitem ___________________________________________________________
cupy/testing/parameterized.py:60: in wrap
    six.raise_from(e_new.with_traceback(e.__traceback__), None)
cupy/testing/parameterized.py:43: in wrap
    return method(*args, **kwargs)
cupy/testing/helper.py:627: in test_func
    impl(self, *args, **kw)
cupy/testing/helper.py:177: in test_func
    check_func(cupy_result, numpy_result)
cupy/testing/helper.py:414: in check_func
    array.assert_array_equal(x, y, err_msg, verbose, strides_check)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x = array([[[ 0.,  1.,  2.,  3.],
        [ 4.,  5.,  6.,  7.],
        [ 8.,  9., 10., 11.]],

       [[16., 17., 18., 19.],
        [20., 21., 22., 23.],
        [24., 25., 26., 27.]]])
y = array([[[12., 13., 14., 15.],
        [ 4.,  5.,  6.,  7.],
        [ 8.,  9., 10., 11.]],

       [[28., 29., 30., 31.],
        [20., 21., 22., 23.],
        [24., 25., 26., 27.]]])
err_msg = '', verbose = True, strides_check = False

    def assert_array_equal(x, y, err_msg='', verbose=True, strides_check=False):
        """Raises an AssertionError if two array_like objects are not equal.

        Args:
             x(numpy.ndarray or cupy.ndarray): The actual object to check.
             y(numpy.ndarray or cupy.ndarray): The desired, expected object.
             strides_check(bool): If ``True``, consistency of strides is also
                 checked.
             err_msg(str): The error message to be printed in case of failure.
             verbose(bool): If ``True``, the conflicting values
                 are appended to the error message.

        .. seealso:: :func:`numpy.testing.assert_array_equal`
        """
        numpy.testing.assert_array_equal(
            cupy.asnumpy(x), cupy.asnumpy(y), err_msg=err_msg,
>           verbose=verbose)
E       AssertionError: Parameterized test failed.
E
E       Base test method: TestArrayAdvancedIndexingVectorValue.test_adv_setitem
E       Test parameters:
E         shape: (2, 3, 4)
E         indexes: (slice(None, None, None), [[0, 1], [2, 0]])
E         value: [[[[ 0  1  2  3]
E          [ 4  5  6  7]]
E
E         [[ 8  9 10 11]
E          [12 13 14 15]]]
E
E
E        [[[16 17 18 19]
E          [20 21 22 23]]
E
E         [[24 25 26 27]
E          [28 29 30 31]]]]
E
E       AssertionError:
E       Arrays are not equal
E
E       Mismatch: 33.3%
E       Max absolute difference: 12.
E       Max relative difference: 1.
E        x: array([[[ 0.,  1.,  2.,  3.],
E               [ 4.,  5.,  6.,  7.],
E               [ 8.,  9., 10., 11.]],...
E        y: array([[[12., 13., 14., 15.],
E               [ 4.,  5.,  6.,  7.],
E               [ 8.,  9., 10., 11.]],...

cupy/testing/array.py:93: AssertionError
```

--- a/tests/cupy_tests/core_tests/test_ndarray_adv_indexing.py
+++ b/tests/cupy_tests/core_tests/test_ndarray_adv_indexing.py
@@ -343,10 +343,10 @@ class TestArrayInvalidValueAdvGetitemForOldVersion(unittest.TestCase):
     {'shape': (2, 3, 4),
      'indexes': (slice(None), [[1, 2], [0, -1]],), 'value': 1},
     {'shape': (2, 3, 4),
-     'indexes': (slice(None), slice(None), [[1, 2], [0, 2]]), 'value': 1},
+     'indexes': (slice(None), slice(None), [[1, 2], [0, -1]]), 'value': 1},
     # slice and array
     {'shape': (2, 3, 4),
-     'indexes': (slice(None), slice(1, 2), [[1, 2], [0, 2]]), 'value': 1},
+     'indexes': (slice(None), slice(1, 2), [[1, 2], [0, -1]]), 'value': 1},
     # None and array
     {'shape': (2, 3, 4),
      'indexes': (None, [1, -1]), 'value': 1},
@@ -507,7 +507,7 @@ class TestArrayAdvancedIndexingSetitemScalarValueIndexError(unittest.TestCase):
      'value': numpy.array([1, 2, 3, 4])},
     {'shape': (2, 3, 4), 'indexes': (slice(None), [0, -1]),
      'value': numpy.arange(2 * 2 * 4).reshape(2, 2, 4)},
-    {'shape': (2, 3, 4), 'indexes': (slice(None), [[0, 1], [2, 0]]),
+    {'shape': (2, 5, 4), 'indexes': (slice(None), [[0, 2], [1, -1]]),
      'value': numpy.arange(2 * 2 * 2 * 4).reshape(2, 2, 2, 4)},
     # mask
     {'shape': (2, 3, 4), 'indexes': numpy.random.choice([False, True], (2, 3)),

